### PR TITLE
[Dashboard] Reduce the number of threads from event module

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

Currently, the event module uses the ThreadPoolExecutor which can start num_cpus * 5 threads (<= 3.7) or min(32, num_cpus + 5) threads (>= 3.8). It means we can have 300s of threads in decently big node. Since we don't need this many threads, we reduce the number of threads by manually specifying the default executor (16 threads). It also adds more internal state information of the agent. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — COMMENTED** [review]
Isn't using more than 1 thread at all a bit excessive for this?

**Comment by rkooo567** [comment]
We currently do concurrent read (and I believe the work is doing some IO). https://github.com/ray-project/ray/blob/2ed09c54459cc3f74e2dab13406018698559856c/dashboard/modules/event/event_utils.py#L185

But right now, we only have 2 files to read, so I think you are right 1 should be sufficient. We can consider increasing it in the future if perf becomes an issue (since we have state logged now, it should be discoverable in the future)

File: dashboard/modules/event/event_head.py [hunk 1]
```diff
@@ -56,10 +64,27 @@ async def ReportEvents(self, request, context):
...
             received_events.extend(parse_event_strings(request.event_strings))
-        logger.info("Received %d events", len(received_events))
+        logger.debug("Received %d events", len(received_events))
         self._update_events(received_events)
+        self.total_report_events_count += 1
+        self.total_events_received += len(received_events)
         return event_pb2.ReportEventsReply(send_success=True)
 
+    @dashboard_utils.async_loop_forever(10)
+    async def _periodic_state_print(self):
```
Comment (ericl) [inline dashboard/modules/event/event_head.py:74]: This seems excessive for logging. We expect like maybe a few events a minute at most right? I think we should remove the periodic state print here and elsewhere.

Comment (rkooo567) [reply dashboard/modules/event/event_head.py:74]: I will remove the periodic printing and find a better way to query this. Will keep the data/logic here (just no periodic printing)

**Review by ericl — APPROVED** [review]

**Pull request merged**